### ISSUE TITLE
🐛 Bug Report: Error when we add repo environment variables and secrets in github

### ISSUE THREAD

**Issue by jaalvarezuria** [backstage/backstage#24993]
### 📜 Description

Hi, 
I have a template where I create a repository with step code called publish:

```
    - id: publish
      name: Publish
      action: publish:github
      input:
        repoUrl: ${{ parameters.githubHost }}?owner=${{ parameters.githubOrganization }}&repo=${{ parameters.repositoryName }}
        repoVisibility: private
        defaultBranch: develop
        repoVariables:
          REGISTRY_FOLDER: ${{ parameters.repositoryName }}
          SONAR_SCAN: "true"
          PRISMA_SCAN: "false"
          PRISMA_CODE_SCAN: "false"
```

This step works properly and I have the repository with repoVariables and secrets inside.

After that, I have another step where I create and environment for the repo with the following code:

```
    - id: envPreEnv
      name: Create Pre Environment
      action: github:environment:create
      input:
        repoUrl: ${{ parameters.githubHost }}?owner=${{ parameters.githubOrganization }}&repo=${{ parameters.repositoryName }}
        name: pre
        environmentVariables:
          CLUSTER: ${{ parameters.repositoryName }}
          DB_CONECTOR: ${{ parameters.repositoryName }}
          DB_SCHEMA: ${{ parameters.repositoryName }}
          REPLICAS: ${{ parameters.repositoryName }}
        secrets:
          DB_SECRET_NAME: ${{ parameters.repositoryName }}
          ROLE_TO_ASSUME: ${{ parameters.repositoryName }}
```

And when it´s executed, I have and error in the log

```
66 2024-05-30T15:12:57.988Z Beginning step Create Pre Environment
67 2024-05-30T15:12:59.933Z HttpError: Not Found - https://docs.github.com/rest
68    at /app/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:125:21
69    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
70    at async requestWithGraphqlErrorHandling (/app/node_modules/@octokit/plugin-retry/dist-node/index.js:71:20)
71    at async Job.doExecute (/app/node_modules/bottleneck/light.js:405:18)
```

I´ve removed the code to create environmentVariables and secrets (just the env name) and it works like a charm.
Pat permissions are ok, but I don´t understand why is not working

### 👍 Expected behavior

Create environment with variables and secrets provided

### 👎 Actual Behavior with Screenshots

It gives an error

### 👟 Reproduction steps

-

### 🖥️ Your Environment

```
OS:   Linux 6.1.79-99.164.amzn2023.x86_64 - linux/x64
node: v20.11.1
yarn: 1.22.22
cli:  0.26.6 (installed)
backstage:  1.27.5

Dependencies:
  @backstage/app-defaults                                          1.5.5
  @backstage/backend-app-api                                       0.4.5, 0.5.14, 0.7.5
  @backstage/backend-common                                        0.18.5, 0.19.10, 0.21.7, 0.22.0
  @backstage/backend-defaults                                      0.2.18
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.11
  @backstage/backend-plugin-api                                    0.5.4, 0.6.18
  @backstage/backend-tasks                                         0.5.23
  @backstage/catalog-client                                        1.6.5
  @backstage/catalog-model                                         1.5.0
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.5
  @backstage/cli                                                   0.26.6
  @backstage/config-loader                                         1.8.0
  @backstage/config                                                1.2.0
  @backstage/core-app-api                                          1.12.5
  @backstage/core-compat-api                                       0.2.5
  @backstage/core-components                                       0.12.5, 0.14.7
  @backstage/core-plugin-api                                       1.9.2
  @backstage/e2e-test-utils                                        0.1.1
  @backstage/errors                                                1.2.4
  @backstage/eslint-plugin                                         0.1.8
  @backstage/frontend-plugin-api                                   0.6.5
  @backstage/integration-aws-node                                  0.1.12
  @backstage/integration-react                                     1.1.27
  @backstage/integration                                           1.11.0
  @backstage/plugin-api-docs                                       0.11.5
  @backstage/plugin-app-backend                                    0.3.67
  @backstage/plugin-app-node                                       0.1.18
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.10
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.10
  @backstage/plugin-auth-backend-module-azure-easyauth-provider    0.1.1
  @backstage/plugin-auth-backend-module-bitbucket-provider         0.1.1
  @backstage/plugin-auth-backend-module-cloudflare-access-provider 0.1.1
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.13
  @backstage/plugin-auth-backend-module-github-provider            0.1.15
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.15
  @backstage/plugin-auth-backend-module-google-provider            0.1.15
  @backstage/plugin-auth-backend-module-guest-provider             0.1.4
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.13
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.15
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.11
  @backstage/plugin-auth-backend-module-oidc-provider              0.1.9
  @backstage/plugin-auth-backend-module-okta-provider              0.0.11
  @backstage/plugin-auth-backend                                   0.22.5
  @backstage/plugin-auth-node                                      0.2.19, 0.4.13
  @backstage/plugin-auth-react                                     0.1.2
  @backstage/plugin-catalog-backend-module-github                  0.6.1
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.16
  @backstage/plugin-catalog-backend                                1.22.0
  @backstage/plugin-catalog-common                                 1.0.23
  @backstage/plugin-catalog-graph                                  0.4.5
  @backstage/plugin-catalog-import                                 0.11.0
  @backstage/plugin-catalog-node                                   1.12.0
  @backstage/plugin-catalog-react                                  1.12.0
  @backstage/plugin-catalog                                        1.20.0
  @backstage/plugin-events-node                                    0.3.4
  @backstage/plugin-kubernetes-common                              0.7.6
  @backstage/plugin-kubernetes-react                               0.3.5
  @backstage/plugin-kubernetes                                     0.11.10
  @backstage/plugin-org                                            0.6.25
  @backstage/plugin-permission-backend-module-allow-all-policy     0.1.15
  @backstage/plugin-permission-backend                             0.5.42
  @backstage/plugin-permission-common                              0.7.13
  @backstage/plugin-permission-node                                0.7.29
  @backstage/plugin-permission-react                               0.4.22
  @backstage/plugin-proxy-backend                                  0.4.16
  @backstage/plugin-scaffolder-backend-module-azure                0.1.10
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.8
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.8
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.8
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.10
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.8
  @backstage/plugin-scaffolder-backend-module-github               0.2.8
  @backstage/plugin-scaffolder-backend-module-gitlab               0.4.0
  @backstage/plugin-scaffolder-backend                             1.22.7
  @backstage/plugin-scaffolder-common                              1.5.2
  @backstage/plugin-scaffolder-node                                0.4.4
  @backstage/plugin-scaffolder-react                               1.8.6
  @backstage/plugin-scaffolder                                     1.20.1
  @backstage/plugin-search-backend-module-catalog                  0.1.24
  @backstage/plugin-search-backend-module-techdocs                 0.1.23
  @backstage/plugin-search-backend-node                            1.2.23
  @backstage/plugin-search-backend                                 1.5.9
  @backstage/plugin-search-common                                  1.2.11
  @backstage/plugin-search-react                                   1.7.11
  @backstage/plugin-search                                         1.4.11
  @backstage/plugin-techdocs-backend                               1.10.5
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.10
  @backstage/plugin-techdocs-node                                  1.12.4
  @backstage/plugin-techdocs-react                                 1.2.4
  @backstage/plugin-techdocs                                       1.10.5
  @backstage/plugin-user-settings                                  0.8.6
  @backstage/release-manifests                                     0.0.11
  @backstage/repo-tools                                            0.9.0
  @backstage/test-utils                                            1.5.5
  @backstage/theme                                                 0.2.19, 0.5.5
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.8
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by acierto** [comment]
Hey @jaalvarezuria,

can you please provide more information about this issue?

As it depends what kind of values did you use in parameters.
I'm a bit puzzled to see "https://docs.github.com/rest" during the github publish

**Comment by jaalvarezuria** [comment]
Hi @acierto, 
Thanks for you response. It´s very strange because I tried several options and all of them give me the same error. In fact, I tried to create variables and secrets in the same way as you can find in the documentation (http://yourserver/create/actions)
![image](https://github.com/backstage/backstage/assets/84963697/4999ea48-302d-4ec2-a4c5-96f696ea5024)

and the result is the same.
If I remove everything related with variables and secrets (just create an environment), github:environment:create works properly.

Let me know if you need more information, but it´s all I can find in the logs.

Thanks

**Comment by ext-mohamed-el-taier** [comment]
Hello, I can also confirm this issue. But i didnt know you can add Repo Vars as below, thanks for sharing and reporting.

    repoVariables:
      REGISTRY_FOLDER: ${{ parameters.repositoryName }}
      SONAR_SCAN: "true"
      PRISMA_SCAN: "false"
      PRISMA_CODE_SCAN: "false"

**Comment by jaalvarezuria** [comment]
Hi @ext-mohamed-el-taier ,
Repo vars are added using   action: publish:github and environment vars uses action: github:environment:create that it´s different, that is why it´s working for repo var but not for environment vars.
I hope somebody can help us to know what´s happening with action: github:environment:create
Thanks

**Comment by benjdlambert** [comment]
Is this related to #25042?

**Comment by jaalvarezuria** [comment]
@benjdlambert yes, it´s the same problem. What I have done it´s create a github action with github api calls to create variables and secrets and call from bakcstage with a github:actions:dispatch

**Comment by ext-mohamed-el-taier** [comment]
> @benjdlambert yes, it´s the same problem. What I have done it´s create a github action with github api calls to create variables and secrets and call from bakcstage with a github:actions:dispatch

nice, bit scrappy though. Are you feeding the secret your action encrypted?

I am also getting intermittent 404 errors when I try to create Github Environment after Publishing a Repo(but it works 70% of the time), which leads me to think there is a consistency issue with Github API.

**Comment by jaalvarezuria** [comment]
No, I´m feeding the secret with random data and I´m populating it directly in github. What I want to do it´s set all vars and secrets needed to perform actions, as a base template.

Regards.

**Comment by davidrichardsonleit** [comment]
I also get a failure when creating secrets for a GitHub environment.

I am using GitHub App for authentication.

And am trying to create a repo environment (yaml definition is below), with a secret.

But I keep getting the error "Bad request - validation failed due to an improperly encrypted secret".

```yaml
    - id: create-dev-github-environment
      name: create dev github environment
      action: github:environment:create
      input:
        repoUrl: ${{ parameters.repoUrl }}
        name: dev
        deploymentBranchPolicy:
          protected_branches: false
          custom_branch_policies: true
        customBranchPolicyNames:
          - dev
        environmentVariables:
          USER_NAME: dev_cicd_user
        secrets:
          PASSWORD: "this is not the dev password"
        token: ${{ secrets.USER_OAUTH_TOKEN }}
```

**Comment by ext-mohamed-el-taier** [comment]
Hi guys, it seems this issue is related to this https://github.com/backstage/backstage/issues/25042.
The Github Actions plugin is using an Octokit that uses different types of identifiers for Github Repo, which would explain the 404's.

**Comment by benjdlambert** [comment]
I think that this should be fixed now in later releases as we've closed #25042, closing this issue, if it still persists for someone feel free to comment and we can re-open.
